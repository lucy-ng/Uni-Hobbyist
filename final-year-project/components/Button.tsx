/* 
650 Industries, Inc., 2024. Styling a React Native button. [Online] 
Available at: https://docs.expo.dev/ui-programming/react-native-styling-buttons/
[Accessed 14 March 2024].
 */

import { styles } from "./Styles";
import { Pressable, Text } from "./Themed";

export default function Button(props: { onPress: any; title: string }) {
  const { onPress, title } = props;
  return (
    <Pressable
      style={styles.button}
      onPress={onPress}
      lightColor="darkgrey"
      darkColor="rgba(255,255,255,0.8)"
    >
      <Text
        style={styles.buttonText}
        darkColor="darkgrey"
        lightColor="rgba(255,255,255,0.8)"
      >
        {title}
      </Text>
    </Pressable>
  );
}

export function CardButton(props: { onPress: any; title: string }) {
  const { onPress, title } = props;
  return (
    <Pressable
      style={styles.button}
      onPress={onPress}
      darkColor="darkgrey"
      lightColor="darkgrey"
    >
      <Text
        style={styles.buttonText}
        darkColor="darkgrey"
        lightColor="darkgrey"
      >
        {title}
      </Text>
    </Pressable>
  );
}
