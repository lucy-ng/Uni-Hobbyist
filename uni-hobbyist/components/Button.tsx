/* 
650 Industries, Inc., 2024. Styling a React Native button. [Online] 
Available at: https://docs.expo.dev/ui-programming/react-native-styling-buttons/
[Accessed 14 March 2024].
 */

import { Link } from "expo-router";
import { styles } from "./Styles";
import { Pressable, Text } from "./Themed";

export default function Button(props: { onPress: any; title: string }) {
  const { onPress, title } = props;
  return (
    <Pressable
      style={styles.button}
      onPress={onPress}
      lightColor="#8D86C9"
      darkColor="#8D86C9"
    >
      <Text style={styles.buttonText} darkColor="white" lightColor="white">
        {title}
      </Text>
    </Pressable>
  );
}

export function RegisterButton() {
  return (
    <Link href="/RegisterScreen" asChild>
      <Pressable style={styles.button} lightColor="#8D86C9" darkColor="#8D86C9">
        <Text style={styles.buttonText} darkColor="white" lightColor="white">
          Register
        </Text>
      </Pressable>
    </Link>
  );
}
