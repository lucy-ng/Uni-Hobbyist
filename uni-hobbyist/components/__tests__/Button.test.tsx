import renderer from "react-test-renderer";
import Button, { RegisterButton } from "../Button";
import { render, screen } from "@testing-library/react-native";
import { fireEvent } from "react-native-testing-library";

beforeAll(() => {
  jest.resetAllMocks();
});

test("button renders correctly", () => {
  const action = jest.fn();

  const tree = renderer
    .create(<Button onPress={action} title="Action" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("register button renders correctly", () => {
  const tree = renderer.create(<RegisterButton />).toJSON();
  expect(tree).toMatchSnapshot();
});