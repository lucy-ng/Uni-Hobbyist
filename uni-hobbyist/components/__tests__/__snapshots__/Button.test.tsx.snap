// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#8D86C9",
        "marginBottom": 20,
        "marginTop": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "white",
        },
        {
          "fontWeight": "bold",
          "letterSpacing": 0.25,
          "lineHeight": 21,
          "padding": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Action
  </Text>
</View>
`;

exports[`register button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="/RegisterScreen"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="link"
  style={
    [
      {
        "backgroundColor": "#8D86C9",
        "marginBottom": 20,
        "marginTop": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "white",
        },
        {
          "fontWeight": "bold",
          "letterSpacing": 0.25,
          "lineHeight": 21,
          "padding": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Register
  </Text>
</View>
`;
